src/app/CommandHandler.cpp

```diff
@@ -186,7 +186,7 @@ void CommandHandler::DecrementHoldOff()
         return;
     }
 
-    if (!mExchangeCtx->IsGroupExchangeContext() && !mSentStatusResponse)
+    if (mExchangeCtx.Get() != nullptr && !mExchangeCtx->IsGroupExchangeContext() && !mSentStatusResponse)
```

Comment:
```suggestion
    if (mExchangeCtx && !mExchangeCtx->IsGroupExchangeContext() && !mSentStatusResponse)
```
Might also be worth logging something here if `!mExchangeCtx`....

Reply:
Auditing this code, I think we also need null-checks in:
* `CommandHandler::ProcessGroupCommandDataIB`
* `CommandHandler::GetAccessingFabricIndex`, perhaps.  This one is a problem.  We used to guarantee that during processing of the incoming message `mExchangeCtx` could not go null, but now with `ExchangeHolder` it can if a command does something to evict the exchange's session, for example.  @mrjerryjohns

Reply:
Added a log on empty exchange.

Reply:
I spun off https://github.com/project-chip/connectedhomeip/issues/21632 on the further possible changes needed here.